File: javascript/14._Longest _Common_Prefix.js
var longestCommonPrefix = function(strs) {
    res = '';
    for (let i=0; i<strs[0].length; i++) {
        for (s of strs) {
            if (i === s.length || strs[0][i] != s[i])  {
                return res;
            }
        }
        res += strs[0][i];
    }
    return res;
};

output1 = longestCommonPrefix(["flower","flow","flight"]);
output2 = longestCommonPrefix(["dog","racecar","car"]);
console.log(output1, output2);

File: python/14._Longest _Common_Prefix.py
'''
Key:
compare th 1st str with all the rest
'''
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ""
        for i in range(len(strs[0])):  # i in length of first string
            for s in strs:  # traverse 3 strings
                if i == len(s) or strs[0][i] != s[i]:  # if the string is run off, or not equal
                    return res
            res += strs[0][i]  # refresh result
        return res

output1 = Solution().longestCommonPrefix(["flower","flow","flight"])
output2 = Solution().longestCommonPrefix(["dog","racecar","car"])
print(output1, output2)